zilliqa/src/api/eth.rs

```diff
@@ -1039,10 +1039,114 @@ fn blob_base_fee(_params: Params, _node: &Arc<Mutex<Node>>) -> Result<()> {
     Err(anyhow!("API method eth_blobBaseFee is not implemented yet"))
 }
 
+/// Calculates the gas rewards for a block based on specified percentiles.
+///
+/// # Parameters
+/// - `block`: The block for which the rewards are being calculated.
+/// - `reward_percentiles`: A slice of percentiles (0-100) at which to calculate the rewards.
+/// - `transactions_sorted_gas_used`: A vector of gas used by transactions, sorted in ascending order.
+///
+/// # Returns
+/// A vector of rewards corresponding to the specified percentiles.
+fn calculate_rewards(
+    block_gas_used: u64,
+    reward_percentiles: &[f64],
+    sorted_gas_prices: &[u128],
+) -> Vec<f64> {
+    let target_gas_values: Vec<f64> = reward_percentiles
+        .iter()
+        .map(|&percentile| percentile / 100.0 * (block_gas_used as f64))
+        .collect();
+
+    let mut rewards = vec![0.0; reward_percentiles.len()];
+    let mut sum_gas = 0.0;
+    let mut percentile_index = 0;
+
+    for gas_used in sorted_gas_prices.iter().map(|&g| g as f64) {
+        sum_gas += gas_used;
```

Comment:
I don't see how this makes sense. `gas_price` is an amount in ZIL. `gas_used` is an amount of `EvmGas`. I think we are mixing them up here.

I suggest we keep the `EvmGas` newtype for longer to avoid this happening?